qa-tester/assets/src/js/admin-bar/components/build-selector/index.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useMemo, useState } from '@wordpress/element';
+
+/**
+ * External dependencies
+ */
+import {
+	Combobox,
+	ComboboxInput,
+	ComboboxPopover,
+	ComboboxList,
+	ComboboxOption,
+} from '@reach/combobox';
+import '@reach/combobox/styles.css';
+
+/**
+ * Internal dependencies
+ */
+import './style.css';
+
+export function BuildSelector( { buildOption, buildOptions, onOptionSelect } ) {
+	const [ , setSelectedOption ] = useState( buildOption );
+	const [ term, setTerm ] = useState( '' );
+	const results = useLabelMatch( term );
+
+	const handleInputChange = ( event ) => {
+		setTerm( event.target.value );
+	};
+
+	const handleOptionSelect = ( buildLabel ) => {
+		const newOption = buildOptions.find(
+			( option ) => option.label === buildLabel
+		);
+		setSelectedOption( newOption );
+		onOptionSelect( newOption );
+	};
+
+	function useLabelMatch( buildLabel ) {
+		return useMemo(
+			() =>
+				buildLabel.trim() === ''
+					? buildOptions.slice( 0, 5 ) // Show the first 5 options by default.
+					: buildOptions.filter( ( option ) =>
+							option.label
+								.toLowerCase()
+								.includes(
+									buildLabel.trim().toLocaleLowerCase()
+								)
+					  ),
+			[ buildLabel ]
+		);
+	}
```

Comment:
I don't think the custom hook is needed.

```suggestion
const results = useMemo(
		() =>
			term === ''
				? buildOptions.slice( 0, 5 ) // Show the first 5 options by default.
				: buildOptions.filter( ( option ) =>
						option.label
							.toLowerCase()
							.includes( term )
				  ),
		[ term ]
	);
```